### PR TITLE
Fix bug(#38077) that causes a stale SA token is being used in istio-cni

### PR DESCRIPTION
**Pull request by srggavrilov**
The BoundServiceAccountTokenVolume Kubernetes feature (graduated to stable in 1.22) improves security of service account tokens by requiring a one hour expiry time, over the previous default of no expiration. This means that install-cni should refetch service account tokens periodically.

This PR integrates periodic SA token checks into already existing sleepCheckInstall function and compatible with legacy token secret mounts.

### CODE REVIEW

File: cni/pkg/install/install.go [hunk 1]
```diff
@@ -273,3 +278,19 @@ func checkInstall(cfg *config.InstallConfig, cniConfigFilepath string) error {
 	}
 	return nil
 }
+
+// watchSAToken periodically reads SA token file and compares its content with the token stored in the Installer.
+// Sends true into fileModified in case of mismatch.
+// Allows to detect changes in the Bound Service Account Token Volume.
+func (in *Installer) watchSAToken(fileModified chan bool, errChan chan error) {
```
Comment (howardjohn) [inline cni/pkg/install/install.go:285]: Seems like this goroutine leaks and is never cleaned up?

**Review by GregHanson — APPROVED** [review]
Release note looks good

**Comment by lei-tang** [comment]
The following code in https://github.com/istio/istio/pull/39801 seems to be rather tricky. If the ctx.Done() is not triggered, the goroutine created by each call of watchSAToken() will never return and cause memory leak.

func (in *Installer) watchSAToken(ctx context.Context, fileModified chan bool, errChan chan error) {
        curToken := in.saToken
        go func() {
                for {
                        select {
                        case <-ctx.Done():
                                return
                        default:
                                token, err := readServiceAccountToken(in.saTokenFilepath)
                                if err != nil {
                                        errChan <- err
                                }
                                if curToken != token {
                                        fileModified <- true
                                }
                                time.Sleep(1 * time.Minute)
                        }
                }
        }()
}

**Pull request merged**